AC3.2-MidUnit-6-Exam/DoubleVideoViewController.swift

```diff
@@ -26,6 +43,104 @@ class DoubleVideoViewController: UIViewController, CellTitled {
     }
     
     @IBAction func loadVideo(_ sender: UIButton) {
+        showImagePickerForSourceType(sourceType: .photoLibrary, fromButton: sender)
+    }
     
+    private func showImagePickerForSourceType(sourceType: UIImagePickerControllerSourceType, fromButton button: UIButton) {
+        let imagePickerController = UIImagePickerController()
+        imagePickerController.modalPresentationStyle = .currentContext
+        imagePickerController.sourceType = sourceType
+        imagePickerController.mediaTypes = [String(kUTTypeMovie)]
+        imagePickerController.delegate = self
+        imagePickerController.modalPresentationStyle = (sourceType == .camera) ? .fullScreen : .popover
+        imagePickerController.videoMaximumDuration = 60
+        
+        self.imagePickerController = imagePickerController
+        self.present(imagePickerController, animated: true, completion: nil)
     }
+    
+    
+    //MARK: - UIImagePickerControllerDelegateMethod
+    func imagePickerController(_ picker: UIImagePickerController,
+                               didFinishPickingMediaWithInfo info: [String : Any]) {
+        switch info[UIImagePickerControllerMediaType] as! String {
+        case String(kUTTypeMovie):
+            if let url = info[UIImagePickerControllerReferenceURL] as? URL {
+                if self.movieURLTop == nil {
```

Comment:
This worked and I gave full credit for functionality, but your solution took more code than it needed and overall I took off 1/2 point (of 2) for this under "Quality and Style", the only (fractional) points you lost.